README.md

```diff
@@ -204,103 +207,57 @@ Assume we have space with `id = 512` and following format on the server:
-To parse particular value, we should define `Value()` method. First two arguments
-are common and unused as a rule, but the third - defines parsed value. So in
-case of POD stuctures it's enough to provide byte-to-byte copy. Since in our
-schema there are fields of three different types, let's descripe three `Value()`
-functions:
-```
-struct UserTupleValueReader : mpp::DefaultErrorHandler {
-    /* Store instance of tuple to be parsed. */
-    UserTuple& tuple;
-    /* Enumerate all types which can be parsed. Otherwise */
-    static constexpr mpp::Type VALID_TYPES = mpp::MP_UINT | mpp::MP_STR | mpp::MP_DBL;
-    UserTupleValueReader(UserTuple& t) : tuple(t) {}
-
-    /* Value's extractors. */
-    void Value(const BufIter_t&, mpp::compact::Type, uint64_t u)
-    {
-       tuple.field1 = u;
-    }
-    void Value(const BufIter_t&, mpp::compact::Type, double d)
-    {
-        tuple.field3 = d;
-    }
-    void Value(const BufIter_t& itr, mpp::compact::Type, mpp::StrValue v)
-    {
-        BufIter_t tmp = itr;
-        tmp += v.offset;
-        std::string &dst = tuple.field2;
-        while (v.size) {
-            dst.push_back(*tmp);
-            ++tmp;
-            --v.size;
-        }
-    }
-};
-```
-It is worth mentioning that tuple itself is wrapped into array, so in fact
-firstly we should parse array. Let's define another one reader:
-```
-template <class BUFFER>
-struct UserTupleReader : mpp::SimpleReaderBase<BUFFER, mpp::MP_ARR> {
-    mpp::Dec<BUFFER>& dec;
-    UserTuple& tuple;
-
-    UserTupleReader(mpp::Dec<BUFFER>& d, UserTuple& t) : dec(d), tuple(t) {}
-    void Value(const iterator_t<BUFFER>&, mpp::compact::Type, mpp::ArrValue)
-    {
-        dec.SetReader(false, UserTupleValueReader{tuple});
-    }
-};
-```
-`SetReader();` sets the reader which is invoked while every entry of the array is
-parsed. Now, to make these two readers work, we should create decoder, set
-its iterator to the position of encoded tuple and invoke `Read()` method:
-```
-    UserTuple tuple;
-    mpp::Dec dec(conn.getInBuf());
-    dec.SetPosition(*t.begin);
-    dec.SetReader(false, UserTupleReader<BUFFER>{dec, tuple});
-    dec.Read();
+
+Member `mpp` is neccessary - it sets the relationship between the structure
+members and associated tuple's fields. It is used by encoder and decoder
+for Object <-> MsgPack serialization. For instance, such structure will be
+serialied as a MsgPack array `[<field1>, <field2>, <field3>]`.
+
+Let's get back to the example with `select`. Consider the request successful.
+We can decode data in this way:
+
+```c++
+if (response.body.data != std::nullopt) {
+    std::vector<UserTuple> results;
+    bool ok = response.body.data->decode(results);
+    if (ok)
+        print_results(results);
+}
 ```
 
+Firstly, we check if the response actually contains any data (Tarantool has
+sent `IPROTO_DATA` in response). According to `IPROTO`, key `IPROTO_DATA`
```

Comment:
What is `IPROTO`? Maybe the `IPROTO` protocol?

Reply:
Good idea, thanks, done.